pkgs/development/interpreters/python/hooks/fix-python-deps.sh

```diff
@@ -0,0 +1,36 @@
+# shellcheck shell=bash
+
+# Setup hook that fixes Python dependencies versions.
+#
+# Example usage in a derivation:
+#
+#   { …, pythonPackages, … }:
+#
+#   pythonPackages.buildPythonPackage {
+#     …
+#     nativeBuildInputs = [ pythonPackages.fixPythonDepsHook ];
+#
+#     # Point to the requirements.txt file, defaults to requirements.txt
+#     requirementsTxt = "requirements_dev.txt";
+#
+#     # This will relax the dependency restrictions
+#     # e.g.: abc>1,<=2 -> abc
+#     relaxPythonDeps = [ "abc" ];
+#     # This will remove the dependency
+#     # e.g.: cde>1,<=2 -> <nothing>
+#     removePythonDeps = [ "cde" ];
+#     …
+#   }
+
+export requirementsTxt
+export relaxPythonDeps
+export removePythonDeps
+export NIX_DEBUG
+
+fixPythonDepsHook() {
+    echo "Executing fixPythonDepsHook"
+
+    @pythonInterpreter@ @fixPythonDepsHookPy@
```

Comment:
```suggestion
fixPythonDepsHook() {
    echo "Executing fixPythonDepsHook"

    export requirementsTxt
    export relaxPythonDeps
    export removePythonDeps
    export NIX_DEBUG
    @pythonInterpreter@ @fixPythonDepsHookPy@
```

or 

```suggestion
fixPythonDepsHook() {
    echo "Executing fixPythonDepsHook"

    export requirementsTxt relaxPythonDeps removePythonDeps NIX_DEBUG
    @pythonInterpreter@ @fixPythonDepsHookPy@
```

Reply:
Sure.

I will probably rename this file again though. I was thinking how I am going to implement it for `setup.py` and my conclusion is that it is better to be 2 different Python scripts (in the same build hook).